lualib/lua_scanners/avast_rest.lua

```diff
@@ -0,0 +1,354 @@
+        if retransmits > 0 then
+
+          retransmits = retransmits - 1
+
+          lua_util.debugm(rule.name, task,
+              '%s: Request Error: %s - retries left: %s',
+              rule.log_prefix, http_err, retransmits)
+
+          -- Select a different upstream!
+          upstream = rule.upstreams:get_upstream_round_robin()
+          addr = upstream:get_addr()
+          request_data.url = make_url(addr, filename)
+
+          lua_util.debugm(rule.name, task, '%s: retry IP: %s:%s',
+              rule.log_prefix, addr, addr:get_port())
+
+          http.request(request_data)
+        else
+          rspamd_logger.errx(task, '%s: failed to scan, maximum retransmits '..
+              'exceed', rule.log_prefix)
+          task:insert_result(rule.symbol_fail, 0.0, 'failed to scan and '..
+              'retransmits exceed')
+        end
+      end
+
+      if http_err then
+        requery()
+      else
+        -- Parse the response
+        if upstream then upstream:ok() end
+
+        local cached = {}
+
+        if code ~= 200 then
+          rspamd_logger.errx(task, 'invalid HTTP code: %s, body: %s, headers: %s', code, body, headers)
+          common.yield_result(task, rule, string.format('Bad HTTP code: %s', code), 1.0, 'fail', maybe_part)
+          return
+        end
+        local data = string.gsub(tostring(body), '[\r\n%s]$', '')
+        lua_util.debugm(rule.name, task, '%s: got reply data: "%s"',
+            rule.log_prefix, data)
+
+        local ucl_parser = ucl.parser()
+        local ok, ucl_err = ucl_parser:parse_string(tostring(body))
+        if not ok then
+          rspamd_logger.errx(task, "%s error parsing json response, retry: %s",
+            rule.log_prefix, ucl_err)
+        end
+        local result = ucl_parser:get_object()
+        lua_util.debugm(N, task, '%s: JSON OBJECT - %s', rule.log_prefix, result)
+
+        local threat_tbl = {}
+        if result and result.issues and #result.issues > 0 then
+          for _,r in ipairs(result.issues) do
+            lua_util.debugm(N, task, '%s: found issue - %s', rule.log_prefix, r)
+            if r.virus then
+              threat_tbl[r.virus] = true
+            elseif r.detections and type(r.detections) == 'table' and #r.detections > 0 then
+              for _,d in ipairs(r.detections) do
+                if d.virus then
+                  threat_tbl[d.virus] = true
+                end
+              end
+            end
+
+            if r.warning_id then
+              -- 42056 - msg: Archive is password protected
+              -- 42110 - msg: The file is a decompression bomb
+              -- 42125 - msg: ZIP archive is corrupted
+              -- 42140 - msg: ISO archive is corrupted
+              -- 42144 - msg: OLE archive is corrupted
+
+              if tostring(r.warning_id) == '42056' then
+                rspamd_logger.warnx(task, '%s: PASSWORD warning: id: %s - msg: %s',
```

Comment:
It must not be on `warning` level conceptually, the maximum logging level for such a message is expected to be `messagex`.